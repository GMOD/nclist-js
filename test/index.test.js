import { fetch } from 'jest-fetch-mock'
import NCListStore from '../src/feature_store'

test('one', async () => {
  const store = new NCListStore({
    baseUrl: '/foo',
    urlTemplate: 'Genes/{refseq}/trackData.json',
    fetch,
  })

  const features = []
  for await (const feature of store.getFeatures({
    refName: 'ctgA',
    start: 0,
    end: 50000,
  })) {
    features.push(feature)
  }

  expect(features.length).toBe(20)
  expect(features).toMatchSnapshot()
})
