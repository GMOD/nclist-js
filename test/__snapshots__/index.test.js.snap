// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`read with generic-filehandle RemoteFile 1`] = `
Array [
  Array [
    0,
    1049,
    9000,
    1,
    "EDEN",
    "EDEN",
    "protein kinase",
    "ctgA",
    "example",
    Array [
      Array [
        0,
        1299,
        9000,
        1,
        "EDEN.3",
        "EDEN.3",
        "Eden splice form 3",
        "ctgA",
        "example",
        Array [
          Array [
            1,
            6999,
            7600,
            1,
            "1",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            4999,
            5500,
            1,
            "1",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            2,
            7600,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
          Array [
            2,
            2999,
            3300,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            2,
            1299,
            1500,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            1,
            3300,
            3902,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
        ],
        "mRNA",
      ],
      Array [
        0,
        1049,
        9000,
        1,
        "EDEN.1",
        "EDEN.1",
        "Eden splice form 1",
        "ctgA",
        "example",
        Array [
          Array [
            2,
            7608,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
          Array [
            1,
            2999,
            3902,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            4999,
            5500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            2,
            1049,
            1200,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            1,
            6999,
            7608,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            1200,
            1500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
        ],
        "mRNA",
      ],
      Array [
        0,
        1049,
        9000,
        1,
        "EDEN.2",
        "EDEN.2",
        "Eden splice form 2",
        "ctgA",
        "example",
        Array [
          Array [
            1,
            6999,
            7608,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            4999,
            5500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            1200,
            1500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            2,
            1049,
            1200,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            2,
            7608,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
        ],
        "mRNA",
      ],
    ],
    "gene",
  ],
]
`;

exports[`read with generic-filehandle remotefile with file urls 1`] = `
Array [
  Array [
    0,
    1049,
    9000,
    1,
    "EDEN",
    "EDEN",
    "protein kinase",
    "ctgA",
    "example",
    Array [
      Array [
        0,
        1299,
        9000,
        1,
        "EDEN.3",
        "EDEN.3",
        "Eden splice form 3",
        "ctgA",
        "example",
        Array [
          Array [
            1,
            6999,
            7600,
            1,
            "1",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            4999,
            5500,
            1,
            "1",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            2,
            7600,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
          Array [
            2,
            2999,
            3300,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            2,
            1299,
            1500,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            1,
            3300,
            3902,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
        ],
        "mRNA",
      ],
      Array [
        0,
        1049,
        9000,
        1,
        "EDEN.1",
        "EDEN.1",
        "Eden splice form 1",
        "ctgA",
        "example",
        Array [
          Array [
            2,
            7608,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
          Array [
            1,
            2999,
            3902,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            4999,
            5500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            2,
            1049,
            1200,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            1,
            6999,
            7608,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            1200,
            1500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
        ],
        "mRNA",
      ],
      Array [
        0,
        1049,
        9000,
        1,
        "EDEN.2",
        "EDEN.2",
        "Eden splice form 2",
        "ctgA",
        "example",
        Array [
          Array [
            1,
            6999,
            7608,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            4999,
            5500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            1,
            1200,
            1500,
            1,
            "0",
            "ctgA",
            "example",
            "CDS",
          ],
          Array [
            2,
            1049,
            1200,
            1,
            "ctgA",
            "example",
            "five_prime_UTR",
          ],
          Array [
            2,
            7608,
            9000,
            1,
            "ctgA",
            "example",
            "three_prime_UTR",
          ],
        ],
        "mRNA",
      ],
    ],
    "gene",
  ],
]
`;
